consensus/spos/commonSubround/subroundStartRound.go

```diff
@@ -203,8 +207,18 @@ func (sr *SubroundStartRound) indexRoundIfNeeded(pubKeys []string) {
 		return
 	}
 
+	currentHeader := sr.Blockchain().GetCurrentBlockHeader()
```

Comment:
Just a thought, the GetCurrentBlockHeader could return the GetGenesisHeader if it is set to nil. Should be refactored if needed in another PR.

Reply:
Right that would simplify code for the caller